test/kubernetes_unittest.cc

```diff
@@ -0,0 +1,71 @@
+#include "../src/configuration.h"
+#include "../src/kubernetes.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class KubernetesTest : public ::testing::Test {
+ protected:
+  MetadataUpdater::ResourceMetadata GetNodeMetadata(
+      const KubernetesReader& reader, const json::Object *node,
+      Timestamp collected_at, bool is_deleted) const
+      throw(json::Exception) {
+    return reader.GetNodeMetadata(node, collected_at, is_deleted);
+  }
+};
+
+TEST_F(KubernetesTest, GetNodeMetadata) {
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataIngestionRawContentVersion: TestVersion\n"
+    "InstanceResourceType: gce_instance\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);
```

Comment:
Let's add a comment with what we're nulling out here -- makes it easier to understand.